pkg/kube/kube_handler.go

```diff
@@ -241,6 +249,7 @@ type handler interface {
 type AuthHandler interface {
 	AsUser(token, cluster string) (handler, error)
 	AsSVC(cluster string) (handler, error)
+	GetOptions() kubeOptions
```

Comment:
I pass the kubeops arguments via handler but I'm not completely happy about the solution this way. 
Is there any better way to have access to the arguments in `GetNamespaces(...)`?

Reply:
@absoludity probably a good Q for you.

Reply:
I'll take a look first thing next week.

Reply:
From a brief look (and note, this is not thought through, but may be worth thinking through), you may not need to pass the arguments themselves down, if you can instead pass down an implementation of an interface which does what you want. That is:

1. Rather than adding the two fields to `handler.Options` you could instead create an interface `NamespaceFetcher` that has the method you require
2. Add that interface as a field on `handler.Options` and have a default implementation that does the current default behavior
3. Add an implementation which does your specific regex implementation, so that it can be instantiated in the main.go when those fields are provided.
4. Pass the interface down to the handlers and just use it.

Again, don't invest effort in this approach unless you're confident it makes sense and would be worthwhile (I've only thought briefly about it). I'll look more closely next week.

Reply:
> I pass the kubeops arguments via handler but I'm not completely happy about the solution this way.
> Is there any better way to have access to the arguments in `GetNamespaces(...)`?

So no, not without refactoring quite a bit which I don't expect you to do. I think having them in the handler is the simplest way (and unfortunately you need the `GetOptions()` accessor rather than using a public field since the handler itself is private isome places). So +1 for method you chose.